smf/clickhouse/method_script.sh

```diff
@@ -28,13 +28,78 @@ ipadm show-addr "$DATALINK/ll" || ipadm create-addr -t -T addrconf "$DATALINK/ll
 ipadm show-addr "$DATALINK/omicron6"  || ipadm create-addr -t -T static -a "$LISTEN_ADDR" "$DATALINK/omicron6"
 route get -inet6 default -inet6 "$GATEWAY" || route add -inet6 default -inet6 "$GATEWAY"
 
-args=(
-  "--log-file" "/var/tmp/clickhouse-server.log"
-  "--errorlog-file" "/var/tmp/clickhouse-server.errlog"
-  "--"
-  "--path" "${DATASTORE}"
-  "--listen_host" "$LISTEN_ADDR"
-  "--http_port" "$LISTEN_PORT"
-)
-
-exec /opt/oxide/clickhouse/clickhouse server "${args[@]}" &
+# Retrieve addresses of the other clickhouse nodes, order them and assign them to be a replica node.
+#
+# TODO: This should probably be 3 replicas
+CH_ADDRS="$(/opt/oxide/internal-dns-cli/bin/dnswait clickhouse -H \
+    | head -n 2 \
+    | tr '\n' ,)"
+
+if [[ -z "$CH_ADDRS" ]]; then
+    printf 'ERROR: found no addresses for other ClickHouse nodes\n' >&2
+    exit "$SMF_EXIT_ERR_CONFIG"
+fi
+
+readarray -td, nodes <<<"$CH_ADDRS,"; declare -p nodes
+
+# Assign addresses to replicas
+REPLICA_HOST_01="${nodes[0]}"
+REPLICA_HOST_02="${nodes[1]}"
+
+# Retrieve addresses of the other clickhouse nodes, order them and assign them to be a keeper node.
+#
+# This should probably be 5 keepers?
```

Comment:
Clarifying a little bit -- this isn't really doing assignment, it's basically trying to observe existing DNS records for clickhouse-keepers and use them, right?

Does this operate correctly if not all three can be found?

Reply:
It cannot. I've added an extra check to make sure it has all addresses